workspaces/ui-v2/src/optic-components/pages/diffs/ReviewEndpointDiffPage/ReviewEndpointDiffPage.tsx

```diff
@@ -71,29 +74,66 @@ export const ReviewEndpointDiffPage: FC<ReviewEndpointDiffPageProps> = ({
   });
   const endpointId = getEndpointId({ method, pathId });
 
-  const allDiffsHandled = shapeDiffs.every(
-    (shapeDiff) =>
-      shapeDiff?.diffDescription?.diffHash &&
-      isDiffHandled(shapeDiff.diffDescription.diffHash)
-  );
-  const [currentIndex, setCurrentIndex] = useState(0);
+  const getNextIncompleteDiff = (recentlyCompletedDiff?: string): number => {
+    for (let i = 0; i < shapeDiffs.length; i++) {
+      const shapeDiff = shapeDiffs[i];
+      if (!shapeDiff.diffDescription) {
+        continue;
+      }
+
+      const isRecentlyCompletedDiff =
+        !!recentlyCompletedDiff &&
+        recentlyCompletedDiff === shapeDiff.diffDescription.diffHash;
+      const diffHandled = isDiffHandled(shapeDiff.diffDescription.diffHash);
+      if (!(isRecentlyCompletedDiff || diffHandled)) {
+        return i;
+      }
+    }
+
+    // If all diffs are complete we should stick on the last rendered diff
+    return shapeDiffs.length - 1;
+  };
+
+  const [currentIndex, setCurrentIndex] = useState(getNextIncompleteDiff());
   const [previewCommands, setPreviewCommands] = useState<any[]>([]);
+  const renderedDiff = shapeDiffs[currentIndex];
 
   return (
     <TwoColumnFullWidth
       left={
-        allDiffsHandled ? (
-          <AllDiffsHandled endpoint={endpoint} />
-        ) : (
-          <ReviewableDiffs
+        <>
+          <RenderedDiffHeader
             endpoint={endpoint}
             shapeDiffs={shapeDiffs}
-            previewCommands={previewCommands}
-            setPreviewCommands={setPreviewCommands}
             currentIndex={currentIndex}
             setCurrentIndex={setCurrentIndex}
           />
-        )
+          <DiffCard
+            key={renderedDiff.diffDescription?.diffHash}
+            updatedSpecChoices={(choices) => {
+              setPreviewCommands(renderedDiff.toCommands(choices));
+            }}
+            diffDescription={renderedDiff.diffDescription!}
+            handled={isDiffHandled(renderedDiff.diffDescription!.diffHash)}
```

Comment:
This isn't used in the `DiffCard` component but I think it would be worthwhile to display to users that this is diff has been handled. The user would have to accept a diff, and then press back to see the old diff (something that I don't believe should be _that_ common).

There's a few other optimizations here such as:
- Setting the input state to what the user had already added in the diff review
- Disabling the input until a new change was done
- Displaying that the diff has already been handled - but more changes can be done
- Being able to review + change an already `Done` diff.

I don't think this is super common, but would be nice polish to add - I also think that this would be easier to implement with a global state that modeled core data (and then things were derived from that)